src/v/kafka/server/handlers/alter_partition_reassignments.cc

```diff
@@ -0,0 +1,114 @@
+
+// Copyright 2020 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+#include "kafka/server/handlers/alter_partition_reassignments.h"
+
+#include "cluster/metadata_cache.h"
+#include "cluster/topics_frontend.h"
+#include "cluster/types.h"
+#include "kafka/protocol/errors.h"
+#include "kafka/protocol/schemata/alter_partition_reassignments_request.h"
+#include "kafka/protocol/schemata/alter_partition_reassignments_response.h"
+#include "kafka/server/errors.h"
+#include "kafka/server/fwd.h"
+#include "kafka/server/quota_manager.h"
+#include "model/metadata.h"
+#include "model/namespace.h"
+#include "model/timeout_clock.h"
+
+#include <seastar/core/coroutine.hh>
+#include <seastar/core/sstring.hh>
+
+#include <absl/container/node_hash_map.h>
+
+#include <algorithm>
+#include <chrono>
+#include <iterator>
+#include <vector>
+
+namespace kafka {
+
+template<>
+ss::future<response_ptr> alter_partition_reassignments_handler::handle(
+  request_context ctx, [[maybe_unused]] ss::smp_service_group ssg) {
+    alter_partition_reassignments_request request;
+    request.decode(ctx.reader(), ctx.header().version);
+    log_request(ctx.header(), request);
+    alter_partition_reassignments_response resp;
+
+    if (request.data.topics.empty()) {
+        co_return co_await ctx.respond(std::move(resp));
+    }
+
+    auto timeout_ms = request.data.timeout_ms;
+    resp.data.responses.reserve(request.data.topics.size());
+    kafka::error_code top_level_err;
+
+    for (const reassignable_topic& topic : request.data.topics) {
+        reassignable_topic_response topic_response{.name = topic.name};
+
+        for (const reassignable_partition& partition : topic.partitions) {
+            model::ntp ntp{
+              model::ns{"kafka"}, topic.name, partition.partition_index};
+            if (partition.replicas) {
+                // Convert each node id to a broker shard because
+                // topics_frontend.move_partition_replicas expects
+                // vector<broker_shard>
+                std::vector<model::broker_shard> replicas;
+                std::for_each(
+                  std::cbegin(partition.replicas),
+                  std::cend(partition.replicas),
+                  [&replicas](const model::node_id& node_id) {
+                      replicas.push_back(
+                        model::broker_shard{.node_id = node_id, .shard = 0});
```

Comment:
Need to figure out how to access the correct shard

Reply:
Spoke to @jcsp  about this.
The recommendation is to follow the logic in [allocation_node.h](https://github.com/redpanda-data/redpanda/blob/dev/src/v/cluster/scheduling/allocation_node.h)

Reply:
Also need to use the quota manager for throttle measures in the response

Reply:
>Spoke to @jcsp about this.
The recommendation is to follow the logic in [allocation_node.h](https://github.com/redpanda-data/redpanda/blob/dev/src/v/cluster/scheduling/allocation_node.h)

unclear if this is done

Reply:
Not yet.